src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveDraftAction.java

```diff
@@ -0,0 +1,400 @@
+                    errors.add(String.format(Const.StatusMessages.FEEDBACK_RESPONSES_INVALID_ID, questionIndx));
+                    continue;
+                }
+                
+                responsesRecipients.add(response.recipient);
+                // if the answer is not empty but the recipient is empty
+                if (response.recipient.isEmpty() && !response.responseMetaData.getValue().isEmpty()) {
+                    errors.add(String.format(Const.StatusMessages.FEEDBACK_RESPONSES_MISSING_RECIPIENT, questionIndx));
+                }
+                
+                if (response.responseMetaData.getValue().isEmpty()) {
+                    // deletes the response since answer is empty
+                    saveResponse(response);
+                } else {
+                    response.giver = questionAttributes.giverType.isTeam() ? userTeamForCourse
+                                                                                : userEmailForCourse;
+                    response.giverSection = userSectionForCourse;
+                    responsesForQuestion.add(response);
+                }
+            }
+                    
+            List<String> questionSpecificErrors =
+                    questionDetails.validateResponseAttributes(responsesForQuestion,
+                                                               data.bundle.recipientList.get(qnId).size());
+            errors.addAll(questionSpecificErrors);
+            
+            if (!emailSet.containsAll(responsesRecipients)) {
+                errors.add(String.format(Const.StatusMessages.FEEDBACK_RESPONSE_INVALID_RECIPIENT, questionIndx));
+            }
+            
+            if (errors.isEmpty()) {
+                for (FeedbackResponseAttributes response : responsesForQuestion) {
+                    saveResponse(response);
+                }
+            } else {
+                List<StatusMessage> errorMessages = new ArrayList<StatusMessage>();
+                
+                for (String error : errors) {
+                    errorMessages.add(new StatusMessage(error, StatusMessageColor.DANGER));
+                }
+                
+                statusToUser.addAll(errorMessages);
+                isError = true;
+            }
+            
+        }
+        
+        if (!isError) {
+            statusToUser.add(new StatusMessage(Const.StatusMessages.FEEDBACK_RESPONSES_SAVED, StatusMessageColor.SUCCESS));
+        }
+
+        if (isUserRespondentOfSession()) {
+            appendRespondant();
+        } else {
+            removeRespondant();
+        }
+        
+        String timeStamp = new SimpleDateFormat("HH.mm.ss").format(new Date());
+        
+        FeedbackSubmissionSaveDraftNotificationAjaxPageData lastSubmittedTimeStamp =
+                new FeedbackSubmissionSaveDraftNotificationAjaxPageData(account, timeStamp, isError);
+        
+        return createAjaxResult(lastSubmittedTimeStamp);
+    }
```

Comment:
Use abstraction to write short and simple methods. Some of our `execute` methods are horribly too long. Let's try to make the new ones shorter.